lewis/core/devices.py

```diff
@@ -111,21 +111,25 @@ class DeviceBuilder(object):
     def __init__(self, module):
         self._module = module
 
-        self._device_types = list(get_members(self._module, is_device).values())
-
-        if not self._device_types:
-            # syslog.warning('No devices discovered in module {}.'.format(module.__name__))
-            pass
-        else:
-            pass
-            # syslog.info('Discovered devices in module {}: {}'.format(
-            #    module.__name__, ','.join([x.__name__ for x in self._device_types])))
-
         submodules = get_submodules(self._module)
 
+        # Devices can be in __init__.py, device.py, or any devices/*.py
+        self._device_types = list(get_members(self._module, is_device).values())
+        self._device_types += list(get_members(submodules.get('device'), is_device).values())
```

Comment:
It wouldn't work that simply because __init__.py is a single module and (I just realize we've used incorrect terminology here and in the code) `devices/`, `interfaces/`, etc are packages.

So `discover_something` actually takes a `some_package` parameter, and iterates over the modules in that package to find classes that fit `something`. This wouldn't work with passing in a module.

I don't know, I'll tinker a little more with it... I'm sure there's a way to make this more consistent.